src/credentials/FabricTable.h

```diff
@@ -351,14 +321,53 @@ class ConstFabricIterator
 class DLL_EXPORT FabricTable
 {
 public:
+    class DLL_EXPORT FabricTableDelegate
+    {
+    public:
+        FabricTableDelegate() {}
+        virtual ~FabricTableDelegate() {}
+
+        /**
+         * Gets called when a fabric is deleted, such as on FabricTable::Delete().
+         **/
+        virtual void OnFabricDeletedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric is loaded into Fabric Table from storage, such as
+         * during FabricTable::Init().
+         **/
+        virtual void OnFabricRetrievedFromStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
+
+        /**
+         * Gets called when a fabric in Fabric Table is persisted to storage, such as
+         * on FabricTable::AddNewFabric().
+         **/
+        virtual void OnFabricPersistedToStorage(FabricTable & fabricTable, FabricIndex fabricIndex) = 0;
```

Comment:
It feels slightly odd to codify this idea here that the fabric table is backed by any storage at all.  Some implementations will not have this.  For instance, if a node is not in-band commissionable and storage is provided elsewhere in the infrastructure, the fabric table will not be the source of truth and instead will be injected with the relevant fabrics by external code that will will call `AddNewFabric` as needed.

Could we simplify this by better defining the cache behavior of the fabric table?  For instance, couldn't we declare:

> _The fabric table is a write through cache, and the storage APIs are synchronous; therefore, on return of `AddNewFabric`, we know that fabrics are persisted in those platforms that do have backing storage._

Or are we providing these callbacks to accommodate a storage API that may be async in the future?

Another observation is that it seems this approach is essentially making the fabric table into a pub / sub.  I don't object to that, but again making this conceptually storage-centric seems like it misses the point.  In the cases where there is no backing storage, what is the meaning of these callbacks?  Whatever that meaning is in the more general case where storage may or may not exist, it seems like the methods names should express those more basic, underlying ideas.

Reply:
This is already existing code! I didn't want to change all of the names

Reply:
That is a perfectly acceptable explanation.

Thanks.